### PR TITLE
[Bug] Prevent Relearnable Egg Moves in Fresh Start

### PR DESCRIPTION
**Pull request by xsn34kzx**
## What are the changes?

Fixed the unintended ability to allow starters to remember unlocked egg moves in the `Fresh Start` challenge with the `Memory Mushroom`.

## Why am I doing these changes?

By definition, the `Fresh Start` challenge is supposed remove all advantages gained from previous runs (abilities, egg moves, etc.). As such, having the ability to remember unlocked egg moves, even as a starter, goes against the intended restriction of not having access to egg moves to begin with. 

## What did change?

An additional check was added to the `if` condition in `getLearnableLevelMoves()` to check if `Fresh Start` is active within the `challenge` list.

### Screenshots/Videos

| Before | After |
| --- | --- |
|  mushroomBefore | mushroomAfter |

## How to test the changes?

With the `Challenge` game mode unlocked, use the following override and complete one wave:
```ts
ITEM_REWARD_OVERRIDE: [ "MEMORY_MUSHROOM" ]
```

Only traditionally learnable moves should be accessible when using the `Memory Mushroom` (i.e., no egg moves should be accessible).

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I considered writing automated tests for the issue?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Comment by ben-lear** [comment]
Could we add 1-2 unit tests to get some coverage for this function? If you need assistance with navigating the Party menu in tests I can help out a bit

**Review by torranx — APPROVED** [review]

File: src/field/pokemon.ts [hunk 1]
```diff
@@ -922,7 +923,7 @@ export default abstract class Pokemon extends Phaser.GameObjects.Container {
    */
   getLearnableLevelMoves(): Moves[] {
     let levelMoves = this.getLevelMoves(1, true).map(lm => lm[1]);
-    if (this.metBiome === -1) {
+    if (this.metBiome === -1 && !this.scene.gameMode.challenges.some(c => c.id === Challenges.FRESH_START)) {
```
Comment (frutescens) [inline src/field/pokemon.ts:926]: It would be nice to have a comment above this conditional, but it isn't necessary.

**Review by frutescens — COMMENTED** [review]
LMK your thoughts on adding a comment there, and I'll approve.

File: src/field/pokemon.ts (hunk 1 above)
Comment (xsn34kzx) [reply src/field/pokemon.ts:926]: I was actually thinking about doing this before I got caught up doing stuff, so I'll probably do this and then bump 🫡

**Review by frutescens — APPROVED** [review]

File: src/field/pokemon.ts (hunk 1 above)
Comment (Tempo-anon) [reply src/field/pokemon.ts:926]: Can use `isFreshStartChallenge()` that I define in https://github.com/pagefaultgames/pokerogue/pull/3420/files ?

**Pull request merged**